src/io/XTcpSocket.cpp

```diff
@@ -0,0 +1,193 @@
+#include "XTcpSocket.h"
+
+// platform dependent system headers
+#ifdef _MSC_VER
```

Comment:
I would say `#ifdef MSVC` is preferred for CMake projects https://cmake.org/cmake/help/latest/variable/MSVC.html

Reply:
Ok, fixed

Reply:
Did not work. Only _MSC_VER is set as preprocessor variable: https://cmake.org/cmake/help/latest/variable/MSVC_VERSION.html#variable:MSVC_VERSION